ompi/communicator/comm.c

```diff
@@ -899,6 +930,73 @@ int ompi_comm_split_type (ompi_communicator_t *comm, int split_type, int key,
         return OMPI_SUCCESS;
     }
 
+    if (MPI_COMM_TYPE_HW_GUIDED == global_split_type) {
```

Comment:
We are doing the check twice once making sure all processes provided MPI_COMM_TYPE_HW_GUIDED as the split and then that they requested the same info. We should merge these two tests into a single one, and only check the consistency of the final split type (plus maybe a flag to indicate how it was provided).

Reply:
I'll work on that. I have an idea of how to combine them.

Reply:
I combined those two collectives in the latest push